airbyte-ci/connectors/pipelines/pipelines/gradle.py

```diff
@@ -37,11 +37,14 @@ def __init__(self, context: PipelineContext) -> None:
         super().__init__(context)
 
     @property
-    def connector_java_build_cache(self) -> CacheVolume:
-        # TODO: remove this once we finish the project to boost source-postgres CI performance.
-        # We should use a static gradle-cache volume name.
-        cache_volume_name = hacks.get_gradle_cache_volume_name(self.context, self.logger)
-        return self.context.dagger_client.cache_volume(cache_volume_name)
+    def persistent_cache_volume(self) -> CacheVolume:
+        """This cache volume is for sharing gradle state across all pipeline runs."""
+        return self.context.dagger_client.cache_volume("gradle-persistent-cache")
+
+    @property
+    def connector_transient_cache_volume(self) -> CacheVolume:
+        """This cache volume is for sharing gradle state across tasks within a single connector pipeline run."""
+        return self.context.dagger_client.cache_volume(f"gradle-{self.context.connector.technical_name}-transient-cache")
```

Comment:
I'm concerned that using a "per pipeline" build cache will disable remote incremental build caching that can benefit multiple connectors on different pipeline.
E.G: if a pipeline builds the java base it will seed the cache and next pipelines (on other connectors) will hit the cache and won't rebuild the java base.

An alternative approach to support remote caching of incremental builds would be to use the gradle S3 remote cache.

Reply:
I don't think this concern is applicable now that we have the CDK. The "java base" you mention is the CDK now and when building a connector we just pull the jars. Regarding more aggressive caching see my top-level response.